### PR TITLE
Migration to pathlib

### LINKED ISSUE #112: Some suggestions
Hi there,

I have successfully installed the OpenNFT software and I have run the tutorials (PSC cont, SVM).
Some suggestions:

**Suggestions:**
1. In Paragraph "Real-time exported files not found in watch folder" [here](https://opennft.readthedocs.io/en/latest/troubleshooting.html) there is a mistake. `{Project Name}/{Subject ID}/{NR Run No:03}_{Image Series No:06}_{#:06}.dcm` should be `{Project Name}/{Subject ID}/{NF Run No:03}_{Image Series No:06}_{#:06}.dcm` otherwise an error is raised.
2. In the `testRTexp.py` python code that is used to mimic the real-time acquisition, I had to modify the code because I use a Mac and a) files like .DS_Store were copied, and b) the copying of the files was not correct since files are not ordered in `filelist`. So, without `sorted()` the first scan that is going to be copied, it might not be the correct one (in chronological order).
I suggest the following 2 modifications (sorted(), & (not filename.startswith(".")):):
```
for filename in sorted(filelist):
    src = os.path.join(srcpath, filename)
    if os.path.isfile(src) & (not filename.startswith(".")):
        dst = os.path.join(dstpath, filename)
        shutil.copy(src, dst)
        print(filename)
        sleep(pause_in_sec) # seconds
```

### PR DESCRIPTION
**Pull request by TheAmid**
Update:

- All _os.path_ methods were replaced with equivalent _pathlib_ package methods

Closes #112

### CODE REVIEW

File: opennft/config.py [hunk 1]
```diff
@@ -7,17 +7,17 @@
...
 
 LOG_LEVEL = 'DEBUG'
 
-ROOT_PATH = os.path.abspath(os.path.dirname(__file__))
-UI_PATH = os.path.join(ROOT_PATH, 'ui')
-PLUGIN_PATH = os.path.join(ROOT_PATH, 'plugins')
-OpenNFT_ICON = os.path.join(ROOT_PATH, 'ui', 'images', 'appicon.png')
-MATLAB_FUNCTIONS_PATH = os.path.join(ROOT_PATH, 'matlab')
+ROOT_PATH = Path(__file__).absolute().resolve().parent
+UI_PATH = ROOT_PATH.joinpath('ui')
```
Comment (espdev) [inline opennft/config.py:17]: use `/` instead of joinpatch:
```python
UI_PATH = ROOT_PATH / 'ui'
```

File: opennft/opennft.py [hunk 2]
```diff
@@ -1044,16 +1044,16 @@ def startFilesystemWatching(self):
             ext = ext[-1]
 
         searchString = self.getFileSearchString(self.P['FirstFileNameTxt'], path, ext)
-        path = os.path.dirname(path)
+        path = path.parent
 
-        logger.info('Searching for {} in {}', searchString, path)
+        logger.info('Searching for {} in {}', searchString, str(path))
```
Comment (espdev) [inline opennft/opennft.py:1049]: do not convert to str in this case

File: opennft/opennft.py [hunk 3]
```diff
@@ -1534,17 +1535,21 @@ def onChooseWeightsFile(self):
...
             return
 
-        if not os.path.isfile(fname):
+        fname = Path(fname)
+
+        if not fname.is_file():
             return
-        fname = fname.replace('/', os.path.sep)
+
+        fname = str(fname)
```
Comment (espdev) [inline opennft/opennft.py:1551]: It is not a good way

```
fname = Path(fname)
fname = str(fname)
```

File: opennft/plugin.py [hunk 4]
```diff
@@ -21,8 +20,8 @@ def __init__(self, parent=None):
         self.setWindowModality(Qt.ApplicationModal)
 
         model = QStandardItemModel(self.lvPlugins)
-        for p in [f for f in os.listdir(config.PLUGIN_PATH) if f.endswith('.py')]:
-            plMod = 'opennft.' + os.path.basename(config.PLUGIN_PATH).lower() + '.' + p[:-3]
+        for p in [f.name for f in config.PLUGIN_PATH.iterdir() if f.suffix == '.py']:
```
Comment (espdev) [inline opennft/plugin.py:23]: please use: 
```python
config.PLUGIN_PATH.glob('.py')
```

File: tests/testRTexp.py [hunk 5]
```diff
@@ -10,15 +10,15 @@
 """
 
 
-import os
 import shutil
 from time import sleep
-import glob
+from pathlib import Path
+from pathlib import PurePath
```
Comment (espdev) [inline tests/testRTexp.py:16]: Why do you use PurePath?

File: tests/testRTexp.py [hunk 6]
```diff
@@ -49,22 +49,23 @@
...
         fname = "{0}{1:03d}.dcm".format(mask, fn)
         filelist.append(fname)
 
-for filename in filelist:
-    src = os.path.join(srcpath, filename)
-    if os.path.isfile(src):
-        dst = os.path.join(dstpath, filename)
+for filename in sorted(filelist):
+    src = filename
+    if Path.is_file(src) & (not str(filename).startswith(".")):
```
Comment (espdev) [inline tests/testRTexp.py:67]: please use `and` instead of `&`

**Review by espdev — CHANGES_REQUESTED** [review]
1. Please use `/` instead of `joinpath`
2. There is a lot of conversions `Path` to `str` in the code. Do we really need this?

File: opennft/config.py (hunk 1 above)
Comment (tiborauer) [reply opennft/config.py:17]: I think `joinpath `is more portable. Supporting Windows may not be on the table but we should not exclude it.

Comment (espdev) [reply opennft/config.py:17]: @tiborauer `/` operator works fine on any supported platforms.

```python
Path('aaa') / 'bbb' / 'ccc'
WindowsPath('aaa/bbb/ccc')
```

Comment (tiborauer) [reply opennft/config.py:17]: Thanks, @espdev. I did not know that.

Comment (TheAmid) [reply opennft/config.py:17]: Replaced with `/`

File: opennft/opennft.py (hunk 2 above)
Comment (TheAmid) [reply opennft/opennft.py:1049]: Done.

File: tests/testRTexp.py (hunk 5 above)
Comment (TheAmid) [reply tests/testRTexp.py:16]: I don't know. So I removed it.

File: tests/testRTexp.py (hunk 6 above)
Comment (TheAmid) [reply tests/testRTexp.py:67]: Done.

File: opennft/plugin.py (hunk 4 above)
Comment (TheAmid) [reply opennft/plugin.py:23]: Thanks, fixed.

File: opennft/opennft.py (hunk 3 above)
Comment (TheAmid) [reply opennft/opennft.py:1551]: Fixed that.

**Review by espdev — APPROVED** [review]

**Pull request merged**